tpm2tools/flush_active_handles_test.go

```diff
@@ -0,0 +1,100 @@
+package tpm2tools
+
+import (
+    "crypto/rand"
+    "crypto/rsa"
+    "testing"
+
+    "github.com/google/go-tpm/tpm2"
+
+    "github.com/google/go-tpm-tools/simulator"
+)
+
+const (
+    // How many keys/handles can the simulator contain at once.
+    maxHandles = 3
+)
+
+func loadRandomExternalKey(t *testing.T, simulator *simulator.Simulator) {
+    pk, err := rsa.GenerateKey(rand.Reader, 2048)
+    if err != nil {
+        t.Fatal(err)
+    }
+    public := tpm2.Public{
+            Type:       tpm2.AlgRSA,
+            NameAlg:    tpm2.AlgSHA1,
+            Attributes: tpm2.FlagSign | tpm2.FlagSensitiveDataOrigin | tpm2.FlagUserWithAuth,
+            RSAParameters: &tpm2.RSAParams{
+                Sign: &tpm2.SigScheme{
+                    Alg:  tpm2.AlgRSASSA,
+                    Hash: tpm2.AlgSHA1,
+                },
+                KeyBits:  2048,
+                Exponent: uint32(pk.PublicKey.E),
+                Modulus:  pk.PublicKey.N,
+            },
+        }
+    private := tpm2.Private{
+        Type:      tpm2.AlgRSA,
+        Sensitive: pk.Primes[0].Bytes(),
+    }
+    _, _, err = tpm2.LoadExternal(simulator, public, private, tpm2.HandleNull)
+    if err != nil {
+        t.Fatal(err)
+    }
+}
+    
+
+func TestFlushActiveHandles(t *testing.T) {
+    simulator, err := simulator.Get()
+    if err != nil {
+        t.Fatal(err)
+    }
+    defer simulator.Close()
+
+    // Loads then flushes 1, 2, ...maxHandles handles.
```

Comment:
We only really need one run of FlushAllTransientHandles here

Reply:
It only loads in transient handles. I kind of like looping load then flush. It helped me identify bugs in the past where I returned incorrect numbers of handles.